include/boost/url/impl/decoded_view.hpp

```diff
@@ -0,0 +1,323 @@
+    {
+        return pos_ == other.pos_;
+    }
+
+    bool
+    operator!=(
+        const_iterator const& other) const noexcept
+    {
+        return !(*this == other);
+    }
+};
+
+constexpr
+decoded_view::const_iterator
+decoded_view::
+begin() const noexcept
+{
+    return {p_, n_, plus_to_space_};
+}
+
+
+constexpr
+decoded_view::const_iterator
+decoded_view::
+cbegin() const noexcept
+{
+    return begin();
+}
+
+constexpr
+decoded_view::const_iterator
+decoded_view::
+end() const noexcept
+{
+    return {p_, n_, plus_to_space_, 0};
+}
+
+constexpr
+decoded_view::const_iterator
+decoded_view::
+cend() const noexcept
+{
+    return {p_, n_, plus_to_space_, 0};
+}
+
+BOOST_URL_CXX17_CONSTEXPR inline
+decoded_view::const_reverse_iterator
+decoded_view::
+rbegin() const noexcept
+{
+    return const_reverse_iterator(end());
+}
+
+BOOST_URL_CXX17_CONSTEXPR inline
+decoded_view::const_reverse_iterator
+decoded_view::
+crbegin() const noexcept
+{
+    return rbegin();
+}
+
+BOOST_URL_CXX17_CONSTEXPR inline
+decoded_view::const_reverse_iterator
+decoded_view::
+rend() const noexcept
+{
+    return const_reverse_iterator(begin());
+}
+
+BOOST_URL_CXX17_CONSTEXPR inline
+decoded_view::const_reverse_iterator
+decoded_view::
+crend() const noexcept
+{
+    return rend();
+}
+
+BOOST_URL_CXX17_CONSTEXPR inline
+decoded_view::const_reference
+decoded_view::
+front() const
+{
+    BOOST_ASSERT( !empty() );
+    return *begin();
+}
+
+BOOST_URL_CXX17_CONSTEXPR inline
+decoded_view::const_reference
+decoded_view::
+back() const
+{
+    BOOST_ASSERT( !empty() );
+    return *rbegin();
+}
+
+BOOST_CXX14_CONSTEXPR inline
+decoded_view::size_type
+decoded_view::
+copy(
+    char* dest,
+    size_type count,
+    size_type pos) const
+{
+    if( pos > size() )
+        detail::throw_invalid_argument(
+            BOOST_CURRENT_LOCATION);
+    std::size_t rlen = (std::min)(count, size() - pos);
+    auto first = begin();
+    while (pos--)
+        ++first;
+    auto last = first;
+    while (rlen--)
+        ++last;
+    for (; first != last; ++first, ++dest)
+        *dest = *first;
+    return rlen;
+}
+
+BOOST_CXX14_CONSTEXPR inline
+int
+decoded_view::
+compare(string_view other) const noexcept
+{
+    std::size_t rlen = (std::min)(size(), other.size());
+
+    // int cmp = traits_type::compare(encoded_data(), other.data(), rlen);
+    auto first1 = begin();
+    auto last1 = end();
+    auto first2 = other.begin();
+    auto last2 = other.end();
+    int cmp = 0;
+    for ( ;
+         (first1 != last1) && (first2 != last2);
+         ++first1, ++first2, --rlen)
+    {
+        if (*first1 < *first2)
+        {
+            cmp = -1;
+            break;
+        }
+        else if (*first2 < *first1)
+        {
+            cmp = +1;
+            break;
+        }
+    }
+    if( cmp != 0 )
+        return cmp;
+    if( size() == other.size() )
+        return 0;
+    return size() < other.size()? -1: +1;
```

Comment:
You have to write this out manually (without the ternary operator) for coverage or at least put these two cases on separate lines.